packages/odyssey-react/src/components/Modal/index.tsx

```diff
@@ -0,0 +1,154 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { createContext, useContext, useMemo } from "react";
+import type { FunctionComponent, ReactElement, ReactNode, ReactText } from "react";
+import { createPortal } from "react-dom";
+import Button from "../Button";
+import type { ButtonVariants } from "../Button";
+import { useOid, useCx } from "../../utils"
+
+export type PropsModal = {
+  /**
+   * The modal content, should use the Static components provided by Modal (Modal.Header, Modal.Body and Modal.Footer)
+   */
+  children: ReactElement,
+
+  /**
+   * The modal id attribute. Automatically generated if not provided.
+   */
+  id?: string,
+
+  /**
+   * Determines whether or not the modal is visible.
+   */
+  open: boolean,
+
+  /**
+   * Function invocation when the modal is opened.
+   */
+   onOpen?: () => void,
+
+  /**
+   * Function invocation when the modal is closed.
+   */
+  onClose: () => void,
+}
+
+export type PropsModalHeader = { 
+  children: ReactText
+}
+
+export type PropsModalBody = { 
+  children: ReactNode
+}
+
+export type PropsModalFooter = { 
+  children: ReactNode
+}
+
+export type PropsModalButton = { 
+  close?: boolean,
+  children: ReactText,
+  variant?: ButtonVariants,
+  onClick?: () => void
+}
+
+export type StaticComponents = {
+  Header: FunctionComponent<PropsModalHeader>,
+  Body: FunctionComponent<PropsModalBody>,
+  Footer: FunctionComponent<PropsModalFooter>,
+  Button: FunctionComponent<PropsModalButton>
+}
+
+/**
+ * UI that appears on top of the main content and moves the system into a mode 
+ * requiring user interaction. This dialog disables the main content until the 
+ * user interacts with the modal dialog.
+ * 
+ * @todo OKTA-00000: [React] Modal: Use proper close icon
+ * @todo Determine how to enforce use of only Modal static components as children using TS
+ * 
+ * @component
+ * @example
+ * <Modal open={true} onOpen={()=>{}} onClose={()=>{}}>
+ *  <Modal.Header>Modal Title</Modal.Header>
+ *  <Modal.Body>
+ *    <p>This is the modal content area. It's width is determined based on the amount of content within it.</p>
+ *  </Modal.Body>
+ *  <Modal.Footer>
+ *    <Button variant="clear" close>Cancel</Button>
+ *    <Button onClick={() => {console.log('do something')}}Continue</Button>
+ *  </Modal.Footer>
+ * </Modal>
+ */
+
+export const ModalContext = createContext<{ onClose: () => void}>({ onClose: () => void 0 });
+
+const Modal: FunctionComponent<PropsModal> & StaticComponents = (props) => {
+  const { children, id, open = false, onClose, onOpen } = props
+  const context = useMemo(() => ({ onClose, onOpen }), [onClose, onOpen]);
```

Comment:
having just the callbacks invoked here is a nice simple start - we can add state shadowing prop behavior soon if we need it